src/client.js

```diff
@@ -383,8 +383,14 @@ Client.prototype.more = function(data) {
 		return;
 	}
 	var chan = target.chan;
-	var count = chan.messages.length - (data.count || 0);
-	var messages = chan.messages.slice(Math.max(0, count - 100), count);
+	var index;
+	chan.messages.some((val, i) => {
+		if (`msg-${val.id}` === data.lastId) {
```

Comment:
Remove `msg-` on the client and send an integer. And I think this `more` change can be split into a separate PR, this will make it a bit easier to review.